### ISSUE TITLE
E: Package 'pep8' has no installation candidate during install.sh

### ISSUE THREAD

**Issue by emdneto** [intrig-unicamp/mininet-wifi#536]
OS: Ubuntu 24.04 LTS
Linux mn-wifi 6.7.11-orbstack SMP Sat Mar 30 12:20:36 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

`sudo util/install.sh -Wlnfv`

```
Detected Linux distribution: Ubuntu 24.04 noble amd64
Ubuntu
sys.version_info(major=3, minor=12, micro=3, releaselevel='final', serial=0)
Detected Python (python3) version 3
Installing Mininet dependencies
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package pep8 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'pep8' has no installation candidate
```

Investigate and fix that. ensure mn-wifi  install on ubuntu 24.04 LTS. I'll work on this.

**Comment by emdneto** [comment]
Maybe replace by `python3-pep8` https://packages.ubuntu.com/search?keywords=python3-pep8

**Comment by Jagadeesh-Kalyanshetti** [comment]
I ran into the same issue recently. Here's the solution that worked for me:
1. Open the terminal (Ctrl + Alt + T).
2. If your mininet-wifi is in the home directory then copy-paste this command --> cd mininet-wifi/util 
     (OR) Navigate to mininet-wifi directory in your machine via terminal and then change directory to util.
3. Type --> gedit install.sh
4. Click on the hamburger icon (three horizontal lines) next to the save option.
5. Select 'Find and Replace'.
6. In the 'Find' field, type --> pep8. In the 'Replace' field, type --> pycodestyle.
7. Click 'Replace All' and save the file.
8. Navigate back to the mininet-wifi directory by typing --> cd ..
9. Everything is all set now. Ensure you are in the mininet-wifi directory and verify if the command works  --> sudo util/install.sh -Wlnfv

**Explanation of the Issue and Resolution:**

The pep8 package was previously used to check Python code against the style conventions in PEP 8. It has since been replaced by pycodestyle, which is why pep8 is no longer available as an installation candidate in Ubuntu 23 and higher.

We resolved this by replacing all instances of pep8 in the install.sh file with pycodestyle. This ensures that any functionality related to pep8 required by mininet-wifi is now directed to pycodestyle.

**Comment by Toyin-Osun** [comment]
I have followed the steps above to resolve the issue raised which I also encountered while trying to install mininet-wifi in ubuntu 24.04. Any one with solution

**Comment by s-saderi** [comment]
I had the same problem with Ubuntu 24.04.
**Solution:**
Downgrade to **Ubuntu 22.04**

**Comment by ramonfontes** [comment]
https://github.com/intrig-unicamp/mininet-wifi/commit/3c8030f3b6d89d7dfd5fe45b51f6502eb7fad7fa fixes the issue with Ubuntu 24.04.

**Comment by emdneto** [comment]
@ramonfontes closing this then
